x/supplier/client/cli/tx_create_claim.go

```diff
@@ -24,12 +25,32 @@ func CmdCreateClaim() *cobra.Command {
 		Short: "Broadcast message create-claim",
 		Args:  cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
-			argSessionHeader := new(sessiontypes.SessionHeader)
-			err = json.Unmarshal([]byte(args[0]), argSessionHeader)
+			// fmt.Println("OLSH")
+			// argSessionHeader := &sessiontypes.SessionHeader{
+			// 	ApplicationAddress:      "pokt1mrqt5f7qh8uxs27cjm9t7v9e74a9vvdnq5jva4",
+			// 	SessionStartBlockHeight: 1,
+			// 	SessionId:               "session_id",
+			// 	SessionEndBlockHeight:   5,
+			// 	Service: &sharedtypes.Service{
+			// 		Id: "anvil",
+			// 	},
+			// }
+			// fmt.Println("HERE", argSessionHeader)
+			// cdc := codec.NewProtoCodec(cdctypes.NewInterfaceRegistry())
+			// bz := cdc.MustMarshalJSON(argSessionHeader)
+			// fmt.Println("HERE", hex.EncodeToString(bz))
+
+			// Get the session header
+			cdc := codec.NewProtoCodec(cdctypes.NewInterfaceRegistry())
+			sessionHeaderBz, err := hex.DecodeString(args[0])
 			if err != nil {
 				return err
 			}
-			argRootHash, err := base64.StdEncoding.DecodeString(args[1])
+			sessionHeader := sessiontypes.SessionHeader{}
+			cdc.MustUnmarshalJSON(sessionHeaderBz, &sessionHeader)
+
+			// Get the root hash
+			rootHash, err := hex.DecodeString(args[1])
```

Comment:
I'm assuming the switch to hex was motivated by the fact that base64 includes "/"?

Assuming that's the issue, wdyt about using [base58](https://pkg.go.dev/github.com/btcsuite/btcutil/base58) instead? I think it would be good to minimize the string length. We could even use `Base58Check` (from the btcutil pkg) if we wanted to disambiguate input/encoding (e.g. copypasta) errors.

Reply:
> I'm assuming the switch to hex was motivated by the fact that base64 includes "/"?

Tbh, I think I just overlooked using base64 because it's okay to have a /. Switchinng it now